src/include/function/table/scan_functions.h

```diff
@@ -12,31 +12,17 @@ class FileSystem;
 
 namespace function {
 
-struct KUZU_API BaseScanSharedState : TableFuncSharedState {
-    std::mutex lock;
-
-    virtual uint64_t getNumRows() const = 0;
-};
-
-struct BaseScanSharedStateWithNumRows : BaseScanSharedState {
-    uint64_t numRows;
-
-    explicit BaseScanSharedStateWithNumRows(uint64_t numRows) : numRows{numRows} {}
-
-    uint64_t getNumRows() const override { return numRows; }
-};
-
-struct ScanSharedState : BaseScanSharedStateWithNumRows {
+struct ScanSharedState : public TableFuncSharedState {
```

Comment:
I think this should be FileScanSharedState. Or I don't mind remove this abstraction and just let each file scan implement their own. There is not much logic in this class anyways. Up to u